Logging of exceptions to console, issue #546

alexkravin opened the pull request:
According to #546 

Now exceptions are logged to console output, e.g.:
while mvn clean verify:

```
Exception happened at PATH: com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck
java.lang.NullPointerException
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.isCstyleMultiDimensionalArrayDeclaration(NoWhitespaceAfterCheck.java:294)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.getArrayTypeOrIdentifier(NoWhitespaceAfterCheck.java:183)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.getPreceded(NoWhitespaceAfterCheck.java:139)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.visitToken(NoWhitespaceAfterCheck.java:111)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:449)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:548)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:373)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:207)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:77)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:263)
	at com.puppycrawl.tools.checkstyle.CheckStyleTask.realExecute(CheckStyleTask.java:318)
	at com.puppycrawl.tools.checkstyle.CheckStyleTask.execute(CheckStyleTask.java:262)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:442)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
	at org.apache.maven.plugin.antrun.AntRunMojo.execute(AntRunMojo.java:327)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[checkstyle] /home/asus/sources/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java:450:13: '}' should be alone on a line.
```

romani commented:
First of decrease of code coverage is not acceptable.
Second, try to think how you changed a logic after that update.
Third, look at exception that you posted, can you reproduce that exception and fix that exception ?

alexkravin commented:
1. Lack of UTs isn't a problem
2. If visitToken faces RuntimeException - not only fall with trace, but log some text and then fall with trace
3. What do you mean reproduce? If I sent a log with exception's trace - I have already reproduced it. Fix when? At runtime? No. But sure NPE is fixable with debugging.

romani commented:
> If visitToken faces RuntimeException - not only fall with trace, but log some text and then fall with trace

I think by "trace" you mean stacktrace. Any part of code could cause or experience RuntimeException, but swallow exception, and do not provide any additional info for engineer to investigate and resolve, so why you catch exception there ?

> What do you mean reproduce? If I sent a log with exception's trace - I have already reproduced it. Fix when? At runtime? No. But sure NPE is fixable with debugging.

Imagine that you run Checkstyle over opneJDK and you see that exception. Please give me how you get from that stacktrace information on how to reproduce it.
If we print smth to log file, that information should be enough to create test case easily.

alexkravin commented:
Seems to be I understood you, now message looks like:
```
Exception happened during analyzing: /home/asus/sources/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java:139
java.lang.NullPointerException
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.isCstyleArrayDeclaration(NoWhitespaceAfterCheck.java:277)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.getArrayTypeOrIdentifier(NoWhitespaceAfterCheck.java:197)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.getPreceded(NoWhitespaceAfterCheck.java:139)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.visitToken(NoWhitespaceAfterCheck.java:111)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:453)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:546)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:378)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:207)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:77)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:263)
	at com.puppycrawl.tools.checkstyle.CheckStyleTask.realExecute(CheckStyleTask.java:318)
	at com.puppycrawl.tools.checkstyle.CheckStyleTask.execute(CheckStyleTask.java:262)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:442)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
	at org.apache.maven.plugin.antrun.AntRunMojo.execute(AntRunMojo.java:327)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[checkstyle] /home/asus/sources/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java:372:42: Expected @throws tag for 'NullPointerException'.
```

So there is printing the name and path to file which fetched to exception, line number of file where exception reproduces, next line is name of exception and next is module which caused the exception.

mkordas commented on src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java:
```diff
@@ -362,9 +367,10 @@ else if (TokenTypes.isCommentType(aToken)) {
      */
     private void walk(DetailAST aAST, FileContents aContents
-            , AstState aAstState)
+            , AstState aAstState) throws NullPointerException
```
`throws NullPointerException` doesn't make sense to me - every method in Java can potentially throw NPE

alexkravin replied:
As it is unchecked exception.. yes, you're right.. It makes sense to catch it in one place

mkordas commented on src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java:
```diff
@@ -223,6 +223,11 @@ protected void processFiltered(File aFile, List<String> aLines)
                     this.getClass(), null));
         }
+        catch (final NullPointerException npe) {
+            System.err.println("Exception happened during analyzing: " + aFile.getPath()
```
what if `aFile` currently is null? Then we'll have NPE out of NPE

mkordas replied:
I would use message like `Exception occurred during the analysis of file `

alexkravin replied:
thanks, updated message and removed odd throwing of NPEs.
About 'if ```aFile``` is currently null' - I think current condition in 'catch' block covers such possible case

romani commented:
1)
> catch (final NullPointerException npe) {

Never catch and throw NPE.
I hope in 2015 we will succeed in enforcing all Check that we have over our code and I will not repeat that again an again.

2)
> ":" + npe.getStackTrace()[2].getLineNumber());

no need to print information from stacktrace if you print stacktrace no matter what. That line number will be form Checkstyle code and not from analysed sources.

3) 
As soon a I recover from flu we will have a talk, for now please review all chapters about Exceptions in Effective Java book.

daniilyar commented:
I think, in case of failure that would be good to log not only the full path to file on which exception has been occured (which is missed at the moment), but also the last DetailAst node on which exception has been happened.

Also could we investigate, is there an option to let Eclipse Checkstyle pluign display more info in case of failure than it does currenlty in case of NPEs? E.g, rethrow hadndled NPE wrapped into RuntimeException with adding context-containing message.. Currently in case of NPE eclipse-cs displays a meaningless error at the beginning of erroneous file, but I hope, that we could think about how to make this better

romani commented:
> but also the last DetailAst node on which exception has been happened.

it that case we need to decide on format to print and this will not let you easily see a place of problem.
 
Exception and full path is more than enough as it all that is required to reproduce a problem. Lets keep simple, I even not sure that introduction of logging systems will help, so I postponed all that thoughts for later time.

Changes for Eclipse-CS should be started by its author, as far as I know he does not have for continue development of Checkstyle. If time is found - initiative should come from plugin. 
You can report an issue on plugin to show interest on that problem.

romani commented:
@alexkravin , please postpone any code changes for this issue, and wait till appear in skype.

alexkravin commented:
Updated

romani commented:
Missed item to fix, please reply me to each point as DONE !!!

2)

    ":" + npe.getStackTrace()[2].getLineNumber());

no need to print information from stacktrace if you print stacktrace no matter what. That line number will be form Checkstyle code and not from analysed sources.

romani commented:
>  Utils.getExceptionLogger().debug("Throwable occured.", err);
+ Utils.getExceptionLogger().error("Exception occured during the analysis of file: "
+ + fileName + ":" + err.getStackTrace()[2].getLineNumber(), err);

It is printing the same information twice ?

alexkravin commented:
2) Ok, removed extra data from message
```Utils.getExceptionLogger().debug("Throwable occured.", err);

Utils.getExceptionLogger().error("Exception occured during the analysis of file: "
+ fileName + ":" + err.getStackTrace()[2].getLineNumber(), err);
```
There was printing the same information with different severity level, now I removed first part and left severity level ERROR, so now it looks like:
```
[checkstyle] Running Checkstyle 6.3-SNAPSHOT on 646 files
Jan 08, 2015 3:21:49 PM com.puppycrawl.tools.checkstyle.TreeWalker processFiltered
SEVERE: Throwable occured during the analysis of file: /home/asus/sources/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java
java.lang.NullPointerException
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.isCstyleArrayDeclaration(NoWhitespaceAfterCheck.java:277)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.getArrayTypeOrIdentifier(NoWhitespaceAfterCheck.java:197)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.getPreceded(NoWhitespaceAfterCheck.java:139)
	at com.puppycrawl.tools.checkstyle.checks.whitespace.NoWhitespaceAfterCheck.visitToken(NoWhitespaceAfterCheck.java:111)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:449)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:542)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:374)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:207)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:77)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:263)
	at com.puppycrawl.tools.checkstyle.CheckStyleTask.realExecute(CheckStyleTask.java:318)
	at com.puppycrawl.tools.checkstyle.CheckStyleTask.execute(CheckStyleTask.java:262)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:442)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
	at org.apache.maven.plugin.antrun.AntRunMojo.execute(AntRunMojo.java:327)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[checkstyle] /home/asus/sources/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java:0: Got an exception - java.lang.NullPointerException
```

romani commented:
Do not miss punctuation - " ." . That is very required part of message to clearly see that message is finished. Logs are text file grammar is ignored!

alexkravin commented:
Added dot after message

romani commented:
1)
> - .debug("RecognitionException occured.", re);
+ Utils.getExceptionLogger().error("Recognition exception 

So what is real name of Exception ? 

2) please do substitute ""Recognition exception occured during the analysis"
+ + " of file: " + fileName + " ."" with String.format(errorMessage,expection, filename)

mkordas commented:
`"Throwable occured during the analysis"` - this message is strange for me. Usually it won't be `Throwable`, but some subclass. What about replacing it with more generic `"Problem occurred during the analysis"`?

Please note that occu**rr**ed is spelled with double 'r'.

romani commented:
"Problem" does really say what exactly happen - validation error or exception.

But thanks for feedback, we need to change Throwable to Exception.

alexkravin commented:
Throwable changed to Exception

1) Names are now ok
2) Replaced with String.format in all cases

romani commented:
1) I asked you to have two parameters in message:
String msg = String.format("%s occured during the analysis of file %s .", "RecognitionException",fileName);
 Utils.getExceptionLogger().error(msg);

2)
and resolve copu-paste - "%s occured during the analysis of file %s .".

romani commented:
merged as FF
